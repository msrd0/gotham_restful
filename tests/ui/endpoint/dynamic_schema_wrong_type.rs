#[macro_use]
extern crate gotham_restful;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

fn schema(_: u16) -> String {
	unimplemented!()
}

fn status_codes() -> Vec<u16> {
	unimplemented!()
}

#[read_all(schema = "schema", status_codes = "status_codes")]
async fn read_all() {}

fn main() {}
