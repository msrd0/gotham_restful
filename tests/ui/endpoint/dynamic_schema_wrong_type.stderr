error[E0631]: type mismatch in function arguments
  --> tests/ui/endpoint/dynamic_schema_wrong_type.rs:16:21
   |
8  | fn schema(_: u16) -> String {
   | --------------------------- found signature defined here
...
16 | #[read_all(schema = "schema", status_codes = "status_codes")]
   |                     ^^^^^^^^ expected due to this
   |
   = note: expected function signature `fn(StatusCode) -> _`
              found function signature `fn(u16) -> _`
note: required by a bound in `invoke`
  --> src/lib.rs
   |
   |         F: FnOnce(T) -> U
   |            ^^^^^^^^^^^^^^ required by this bound in `invoke`

error[E0308]: mismatched types
  --> tests/ui/endpoint/dynamic_schema_wrong_type.rs:16:46
   |
16 | #[read_all(schema = "schema", status_codes = "status_codes")]
   |                                              ^^^^^^^^^^^^^^
   |                                              |
   |                                              expected struct `StatusCode`, found `u16`
   |                                              expected due to this
   |
   = note: expected struct `Vec<StatusCode>`
              found struct `Vec<u16>`
